test/e2e/util_test.go

```diff
@@ -946,3 +946,20 @@ func SetupGeneratedTestNamespace(name string) corev1.Namespace {
 
 	return ns
 }
+
+func TeardownNamespace(ns *corev1.Namespace) {
+	log := ctx.Ctx().Logf
+
+	currentTest := CurrentGinkgoTestDescription()
+	if currentTest.Failed {
+		log("collecting the %s namespace artifacts as the '%s' test case failed", ns.GetName(), currentTest.TestText)
+		if err := ctx.Ctx().DumpNamespaceArtifacts(ns.GetName()); err != nil {
+			log("failed to collect namespace artifacts: %v", err)
```

Comment:
We avoid performing operations like `Expect(err).ToBe(Nil())` here as we want the namespace to be deleted in any case, regardless of whether this method fails to gather artifacts.